src/Test/AtomicTests/variables.jl

```diff
@@ -0,0 +1,178 @@
+    @test MOI.get(model, MOI.NumberOfVariables()) == 0
+    v = MOI.addvariables!(model, 2)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 2
+end
+atomictests["add_variables"]    = add_variables
+
+"""
+    This function tests adding, and then deleting,
+    a single variable.
+"""
+function delete_variable(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    @test MOI.isempty(model)
+    @test MOI.canaddvariable(model)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 0
+    v = MOI.addvariable!(model)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 1
+    @test MOI.candelete(model, v)
+    MOI.delete!(model, v)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 0
+end
+atomictests["delete_variable"]  = delete_variable
+
+"""
+    This function tests adding, and then deleting,
+    multiple variables.
+"""
+function delete_variables(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    @test MOI.isempty(model)
+    @test MOI.canaddvariable(model)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 0
+    v = MOI.addvariables!(model, 2)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 2
+    @test MOI.candelete(model, v)
+    MOI.delete!(model, v)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 0
+    v = MOI.addvariables!(model, 2)
+    @test MOI.get(model, MOI.NumberOfVariables()) == 2
+    @test MOI.candelete(model, v[1])
+    MOI.delete!(model, v[1])
+    @test MOI.get(model, MOI.NumberOfVariables()) == 1
+    @test !MOI.candelete(model, v[1])
+    @test MOI.candelete(model, v[2])
+    @test !MOI.isvalid(model, v[1])
+    @test MOI.isvalid(model, v[2])
+end
+atomictests["delete_variables"] = delete_variable
+
+"""
+    Test getting variables by name.
+"""
+function getvariable(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    MOIU.loadfromstring!(model,"""
+        variables: x
+        minobjective: 2.0x
+        c1: x >= 1.0
+        c2: x <= 2.0
+    """)
+    @test MOI.canget(model, MOI.VariableIndex, "x")
+    @test !MOI.canget(model, MOI.VariableIndex, "y")
+    x = MOI.get(model, MOI.VariableIndex, "x")
+    @test MOI.isvalid(model, x)
+end
+atomictests["getvariable"]      = getvariable
+
+"""
+    Test getting and setting varaible names.
+"""
+function variablenames(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    v = MOI.addvariable!(model)
+    @test MOI.get(model, MOI.VariableName(), v) == ""
+    @test MOI.canset(model, MOI.VariableName(), typeof(v))
+    MOI.set!(model, MOI.VariableName(), v, "x")
+    @test MOI.get(model, MOI.VariableName(), v) == "x"
+    MOI.set!(model, MOI.VariableName(), v, "y")
+    @test MOI.get(model, MOI.VariableName(), v) == "y"
+    x = MOI.addvariable!(model)
+    MOI.set!(model, MOI.VariableName(), x, "x")
+    @test MOI.get(model, MOI.VariableName(), x) == "x"
+end
+atomictests["variablenames"]    = variablenames
+
+"""
+    Test the setting of an upper bound
```

Comment:
Another example where "solve" in the name would be informative, e.g., `solve_with_upperbound`.